pdfminer/utils.py

```diff
@@ -17,6 +18,8 @@ class open_filename(object):
     (just like `open`), but does nothing for file-like objects.
     """
     def __init__(self, filename, *args, **kwargs):
+        if isinstance(filename, pathlib.PurePath):
+            filename = str(filename)
         if isinstance(filename, str):
             self.file_handler = open(filename, *args, **kwargs)
             self.closing = True
```

Comment:
While we are changing this code, could you also add a conditional to the file_handler clause? E.g. use `elif` and raise an exception in the `else` clause that this type of input is not supported.

Reply:
Fixed in 023a0e3